### ISSUE TITLE
minimum_cut returns correct value (flow) but wrong partition for some graphs

### ISSUE THREAD

**Issue by harristeague** [networkx/networkx#4972]
I have found a case where calls to minimum_cut() returns the correct "value" but incorrect "partition".  I traced the source of this problem to the following line:

In:
https://networkx.org/documentation/stable/_modules/networkx/algorithms/flow/maxflow.html#minimum_cut

Line:
cutset = [(u, v, d) for u, v, d in R.edges(data=True) if d["flow"] == d["capacity"]]

I have cases where the residual R is returned with some edges where d["flow"] differs from d["capacity"]] by a very, very small number (1e-14 for example).  They are clearly supposed to be equal.  However, they are not included in the cutset thus the partition comes out very wrong.  I have seen this with both dinitz and preflowpush flow functions.

It seems that a "nearly equal" comparison would solve this issue.  I realize this might be dangerous since it's not clear how to set the threshold for nearly equal.  Perhaps there is a more elegant way.  I'm not a contributor so wouldn't know how to submit a fix (sorry).

**Comment by harristeague** [comment]
Here is some code that will read the attached graph and print the result.  Note that the error is not always produced.  I'm not sure why, but you may need to run a few times to get the error to appear.

**Comment by harristeague** [comment]
```
import networkx as nx
import json

file = 'Cut_debug_graph.json'
with open(file, 'r') as fp:
    data = json.load(fp)
graph = nx.node_link_graph(data, multigraph=False, directed=True)
value, partition = nx.minimum_cut(graph, 's', 't', capacity='out_cost')
print(f'Value: {value}\nPartition: {partition}')
```

**Comment by harristeague** [comment]
[Cut_debug_graph.json.txt](https://github.com/networkx/networkx/files/6803395/Cut_debug_graph.json.txt)

**Comment by harristeague** [comment]
It was suggested by @rossbar that I reference #4641 which could be related.

**Comment by harristeague** [comment]
When running the above code, this is the output when the partition is **incorrect**:
```
Value: 373.55
Partition: ({'s'}, {'9', '3', '4', '0.agg', 't', '7', '3.agg'})
```
This is an example of the output when the partition is **correct**:
```
Value: 373.55
Partition: ({'3', '4', '0.agg', '7', 's'}, {'3.agg', '9', 't'})
```

**Comment by jamestrimble** [comment]
@harristeague I agree about what you've said about the cause of the problem, and I'm also unsure about what the best solution would be.

My two cents is that the non-determinism seems to be due to the `arbitrary_element()` call on [this line](https://github.com/networkx/networkx/blob/c47964d7ab849d143cd57607a2a6dbbb53bbaf30/networkx/algorithms/flow/preflowpush.py#L232). This just calls `next(iter(...))` on a set, so it's unsurprising (and probably not worrying) that the answer changes from run to run.

**Comment by jamestrimble** [comment]
#4592 might be related.

**Comment by MridulS** [comment]
We have added a section to our tutorial so users are careful while using floats as edge attribute values https://networkx.org/documentation/latest/tutorial.html#floating-point-considerations
